kratos/utilities/variable_utils.h

```diff
@@ -934,11 +934,21 @@ class KRATOS_API(KRATOS_CORE) VariableUtils
     void UpdateCurrentToInitialConfiguration(const ModelPart::NodesContainerType& rNodes);
 
     /**
-     * @brief This method updates the initial nodal coordinates to the current coordinates
      * @param rNodes the nodes to be updated
+     * @brief This method updates the initial nodal coordinates to the current coordinates
      */
     void UpdateInitialToCurrentConfiguration(const ModelPart::NodesContainerType& rNodes);
 
+    /**
+     * @brief This method updates the current coordinates
+     * For each node, this method takes the value of the provided variable and updates the
+     * current position as the initial position (X0, Y0, Z0) plus such variable value
+     * @param rNodes
+     */
+    void UpdateCurrentPosition(
+        const ModelPart::NodesContainerType &rNodes,
+        const ArrayVarType &rUpdateVariable = DISPLACEMENT);
```

Comment:
```suggestion
        const ArrayVarType& rUpdateVariable = DISPLACEMENT
        );
```

I am quite sure that in our style guide we use type& thing instead of type &thing

Reply:
@philbucher

Reply:
> I am quite sure that in our style guide we use type& thing instead of type &thing

Good practice is to attach the & to the thing that is to be a reference. Indeed, there is a corner case that might yield an unexpected behaviour if not (@jcotela explained it to me once but I can't remember it right now).

Reply:
https://stackoverflow.com/questions/22766907/whats-the-semantically-accurate-position-for-the-ampersand-in-c-references 

> A typical C programmer writes int *p; and explains it *p is what is the int emphasizing syntax, and may point to the C (and C++) declaration grammar to argue for the correctness of the style. Indeed, the * binds to the name p in the grammar.

 >   A typical C++ programmer writes int* p; and explains it p is a pointer to an int emphasizing type. Indeed the type of p is int*. I clearly prefer that emphasis and see it as important for using the more advanced parts of C++ well.

Reply:
As you can see is an open discussion, but the type& thing style is more C++ and is by far the most used on Kratos, and then `de facto` style

Reply:
This is the example that @rubenzorrilla mentioned, for pointers but the same goes for references:
https://coliru.stacked-crooked.com/a/b0422d16be62a272
This is why `int &a, b;` is clearer than `int& a, b;` (the `&` applies only to the first definition).
Most code style guides will tell you to never define multiple variables to the same line anyways, so the point is somewhat moot.

Reply:
Thanks for the clarification @jcotela .